plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/AbstractXmlValidatorTest.java

```diff
@@ -29,58 +33,99 @@
+import org.eclipse.jst.j2ee.web.project.facet.WebFacetUtils;
+import org.eclipse.wst.common.project.facet.core.IProjectFacetVersion;
+import org.eclipse.wst.common.project.facet.core.ProjectFacetsManager;
+import org.eclipse.wst.validation.ValidationFramework;
+import org.eclipse.wst.validation.Validator;
 import org.junit.After;
-import org.junit.BeforeClass;
+import org.junit.Before;
 import org.junit.ClassRule;
 import org.junit.Test;
 import org.xml.sax.SAXException;
 import org.xml.sax.SAXParseException;
 
 public class AbstractXmlValidatorTest {
 
-  private final String ELEMENT_MESSAGE = "Project ID should be specified at deploy time.";
-  private static IResource resource;
-  private static IProject project;
+  private static final IProjectFacetVersion APPENGINE_STANDARD_FACET_VERSION_1 =
+      ProjectFacetsManager.getProjectFacet(AppEngineStandardFacet.ID).getVersion("1");;
+  private IFile resource;
 
-  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator();
+  @ClassRule public static TestProjectCreator appEngineStandardProjectCreator =
+      new TestProjectCreator().withFacetVersions(Lists.newArrayList(JavaFacet.VERSION_1_7,
+          WebFacetUtils.WEB_25, APPENGINE_STANDARD_FACET_VERSION_1));
+  
+  @ClassRule public static TestProjectCreator dynamicWebProjectCreator =
+      new TestProjectCreator().withFacetVersions(Lists.newArrayList(JavaFacet.VERSION_1_7,
+                                                                    WebFacetUtils.WEB_25));
 
-  @BeforeClass
-  public static void setUp() throws CoreException {
-    project = projectCreator.getProject();
+  @Before
+  public void setUp() throws CoreException {
+    IProject project = dynamicWebProjectCreator.getProject();
     createFolders(project, new Path("src/main/webapp/WEB-INF"));
-    IFile webXml = project.getFile("src/main/webapp/WEB-INF/web.xml");
-    webXml.create(new ByteArrayInputStream(new byte[0]), true, null);
-    resource = webXml;
+    resource = project.getFile("src/main/webapp/WEB-INF/web.xml");
+    resource.create(new ByteArrayInputStream(new byte[0]), true, null);
   }
-
+  
   @After
   public void tearDown() throws CoreException {
-    if (resource != null) {
-      resource.deleteMarkers(IMarker.PROBLEM, true, IResource.DEPTH_ZERO);
+    resource.delete(true, null);
+  }
+  
+  @Test
+  public void testValidate_appEngineStandard() throws CoreException {
+    IProject project = appEngineStandardProjectCreator.getProject();
+    createFolders(project, new Path("src/main/webapp/WEB-INF"));
+    IFile file = project.getFile("src/main/webapp/WEB-INF/web.xml");
+    ValidationFramework framework = ValidationFramework.getDefault();
+    Validator[] validators = framework.getValidatorsFor(file);
+    for (Validator validator : validators) {
+      if ("com.google.cloud.tools.eclipse.appengine.validation.webXmlValidator"
+          .equals(validator.getId())) {
+        return;
+      }
     }
+    fail();
```

Comment:
Might be useful to provide a description here: "webXmlValidator missing from AppEngine Standard project"